src/api/Metadata.js

```diff
@@ -926,7 +926,19 @@ class Metadata extends File {
                 const key = this.getMetadataCacheKey(id);
                 const cachedMetadata = cache.get(key);
 
-                const templateInstance = { ...template, type: metadata.data.$type };
+                let templateId;
+                if (cachedMetadata && cachedMetadata.templates) {
+                    const foundTemplate = cachedMetadata.templates.find(
+                        cachedTemplate =>
+                            cachedTemplate.templateKey === template.templateKey &&
+                            cachedTemplate.scope === template.scope,
+                    );
+                    templateId = foundTemplate ? foundTemplate.id : template.id;
+                } else {
+                    templateId = template.id;
+                }
```

Comment:
This could be simplified to:
```
                let foundTemplate;
                if (cachedMetadata && cachedMetadata.templates) {
                    const foundTemplate = cachedMetadata.templates.find(
                        cachedTemplate =>
                            cachedTemplate.templateKey === template.templateKey &&
                            cachedTemplate.scope === template.scope,
                    );
                }
                const templateId = foundTemplate ? foundTemplate.id : template.id;
```
WDYT?

Reply:
Good idea!